pyspecdata/core.py

```diff
@@ -4591,6 +4591,16 @@ def nnls(self, dimname, newaxis_dict, kernel_func, l=0):
             each kernel is returned in properties of the nddata called,
             respectively, "s1" and "s2". 
         """
+        def demand_real(x, addtxt=''):
+            if not x.dtype == real64:
+                if x.dtype == complex128:
+                    raise ValueError("you are not allows to pass nnls complex data:\nif it makes sense for you, try yourdata.real.nnls( where you now have yourdata.nnls("+'\n'+addtxt)
+                else:
+                    raise ValueError("I expect double-precision floating point (float64), but you passed me data of dtype "+str(x.dtype)+'\n'+addtxt)
+        demand_real(self.data)
+        demand_real(self.getaxis(dimname),"(this message pertains to the %s axis)"%dimname)
+        for k,v in newaxis_dict.items():
+            demand_real(v,"(this message pertains to the new %s axis)"%str(k))
```

Comment:
following on your comments, I'm trying to check that _all_ the important things are real